MdeModulePkg/Include/Library/DebugAgentLib.h

```diff
@@ -21,6 +21,8 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 #define DEBUG_AGENT_INIT_DXE_LOAD             10
 #define DEBUG_AGENT_INIT_DXE_UNLOAD           11
 #define DEBUG_AGENT_INIT_THUNK_PEI_IA32TOX64  12
+#define DEBUG_AGENT_INIT_REINITIALIZE         13

```

Comment:
I think DebugAgent library instance is also required to be updated to support new types.

Reply:
And, what usage is for these two new types?

Reply:
DEBUG_AGENT_INIT_REINITIALIZE is to allow the debugger to reinitialize. This was being used after EFI Applications were launched, to allow the debugger to potentially reinitialize after an application may have made changes.

DEBUG_AGENT_INIT_DXE_CORE_LATE is to allow the debugger to initialize after Event Services are available, so that they debugger can register event notifications to break into the system, etc.

Reply:
Will you plan to update DebugAgent library instance to support new type?

Reply:
I pushed a commit with the cases handled to prevent an assert.

That was what you were asking about, correct?